Tests/FluentAssertions.Specs/Primitives/DateOnlyAssertionSpecs.cs

```diff
@@ -0,0 +1,863 @@
+using Xunit.Sdk;
+
+#if NET6_0_OR_GREATER
+
+namespace FluentAssertions.Specs.Primitives
+{
+    public class DateOnlyAssertionSpecs
+    {
+        #region (Not) Be
+
+        [Fact]
+        public void Should_succeed_when_asserting_dateonly_value_is_equal_to_the_same_value()
+        {
+            // Arrange
+            DateOnly dateOnly = new(2016, 06, 04);
+            DateOnly sameDateOnly = new(2016, 06, 04);
+
+            // Act
+            Action act = () => dateOnly.Should().Be(sameDateOnly);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_dateonly_value_is_equal_to_the_same_nullable_value_be_should_succeed()
+        {
+            // Arrange
+            DateOnly dateOnly = new(2016, 06, 04);
+            DateOnly? sameDateOnly = new(2016, 06, 04);
+
+            // Act
+            Action act = () => dateOnly.Should().Be(sameDateOnly);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_both_values_are_at_their_minimum_then_it_should_succeed()
+        {
+            // Arrange
+            DateOnly dateOnly = DateOnly.MinValue;
+            DateOnly sameDateOnly = DateOnly.MinValue;
+
+            // Act
+            Action act = () => dateOnly.Should().Be(sameDateOnly);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_both_values_are_at_their_maximum_then_it_should_succeed()
+        {
+            // Arrange
+            DateOnly dateOnly = DateOnly.MaxValue;
+            DateOnly sameDateOnly = DateOnly.MaxValue;
+
+            // Act
+            Action act = () => dateOnly.Should().Be(sameDateOnly);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void Should_fail_when_asserting_dateonly_value_is_equal_to_the_different_value()
+        {
+            // Arrange
+            var dateOnly = new DateOnly(2012, 03, 10);
+            var otherDateOnly = new DateOnly(2012, 03, 11);
+
+            // Act
+            Action act = () => dateOnly.Should().Be(otherDateOnly, "because we want to test the failure {0}", "message");
+
+            // Assert
+            act.Should().Throw<XunitException>()
+                .WithMessage("Expected dateOnly to be <2012-03-11>*failure message, but found <2012-03-10>.");
+        }
+
+        [Fact]
+        public void Should_succeed_when_asserting_dateonly_value_is_not_equal_to_a_different_value()
+        {
+            // Arrange
+            DateOnly dateOnly = new(2016, 06, 04);
+            DateOnly otherDateOnly = new(2016, 06, 05);
+
+            // Act
+            Action act = () => dateOnly.Should().NotBe(otherDateOnly);
+
+            // Assert
+            act.Should().NotThrow();
+        }
+
+        [Fact]
+        public void When_dateonly_value_is_not_equal_to_a_different_nullable_value_notbe_should_succeed()
```

Comment:
The method body is identical to `Should_succeed_when_asserting_dateonly_value_is_not_equal_to_a_different_value` above.